prusti-viper/src/encoder/foldunfold/log.rs

```diff
@@ -25,7 +26,7 @@ pub(super) struct EventLog {
     /// Actions performed by the fold-unfold algorithm before the join. We can use a single
     /// CfgBlockIndex because fold-unfold algorithms generates a new basic block for dropped
     /// permissions.
-    prejoin_actions: HashMap<vir::CfgBlockIndex, Rc<Vec<Action>>>,
+    prejoin_actions: HashMap<vir::CfgBlockIndex, Rc<RwLock<Vec<Action>>>>,
```

Comment:
This is necessary because after a `PathCtxt` is cloned, the `Rc` forbids write access (because now there are two `Rc`s pointing to the same memory). Is the `Rc` still the right thing to have (instead of a plain `Vec<Action>`)? The way it is now the `then_pctxt` and the `else_pctxt` have the same `prejoin_actions`, which does seem odd.